### PR TITLE
Misc. distributed fixes/tweaks

### PR DESCRIPTION
**Pull request by majosm**
Pulled out of #204 to declutter.

### CODE REVIEW

File: meshmode/mesh/processing.py [hunk 1]
```diff
@@ -447,52 +447,52 @@ def _create_inter_partition_adjacency_groups(mesh, part_per_element,
...
+            neighbor_parts = np.empty(nfaces, dtype=np.int32)
             neighbor_parts.fill(-1)
             neighbors = bdry.neighbors
-            neighbor_elements = np.empty(nelems, dtype=mesh.element_id_dtype)
+            neighbor_elements = np.empty(nfaces, dtype=mesh.element_id_dtype)
             neighbor_elements.fill(-1)
-            neighbor_faces = np.empty(nelems, dtype=mesh.face_id_dtype)
-            neighbor_faces.fill(-1)
+            neighbor_faces = np.empty(nfaces, dtype=mesh.face_id_dtype)
+            neighbor_faces.fill(0)
```
Comment (alexfikl) [inline meshmode/mesh/processing.py:483]:
```suggestion
            neighbor_faces = np.zeros(nfaces, dtype=mesh.face_id_dtype)
```

**Review by alexfikl — COMMENTED** [review]
Left a few nitpicks, hope that's ok!

File: meshmode/mesh/processing.py [hunk 2]
```diff
@@ -447,52 +447,52 @@ def _create_inter_partition_adjacency_groups(mesh, part_per_element,
...
-            neighbors = np.empty_like(elements)
-            for inonlocal in range(len(neighbors)):
-                i_neighbor_part = neighbor_parts[inonlocal]
-                from meshmode.mesh import BTAG_REALLY_ALL, BTAG_PARTITION
-                neighbors[inonlocal] = -(
-                                boundary_tag_bit(BTAG_REALLY_ALL)
-                                | boundary_tag_bit(BTAG_PARTITION(i_neighbor_part)))
+            from meshmode.mesh import BTAG_REALLY_ALL, BTAG_PARTITION
+            flags = np.empty_like(elements)
+            flags[:] = boundary_tag_bit(BTAG_REALLY_ALL)
```
[the remark this answers is not in the document]
Comment (majosm) [reply meshmode/mesh/processing.py:464]: Neat. I didn't know about `full`/`full_like`.

File: meshmode/mesh/processing.py (hunk 1 above)
Comment (majosm) [reply meshmode/mesh/processing.py:483]: 🤦 Thanks!

**Comment by inducer** [comment]
Unsubscribing... @-mention or request review once it's ready for a look or needs attention.

**Comment by majosm** [comment]
I think this one is ready for a look @inducer.

File: meshmode/mesh/__init__.py [hunk 3]
```diff
@@ -1273,6 +1273,8 @@ def as_python(mesh, function_name="make_mesh"):
         # {{{ facial adjacency groups
 
         def fagrp_params_str(fagrp):
+            if isinstance(fagrp, InterPartitionAdjacencyGroup):
+                raise NotImplementedError("Not implemented for distributed meshes.")
```
Comment (inducer) [inline meshmode/mesh/__init__.py:1277]:
```suggestion
            if not isinstance(fagrp, FacialAdjacencyGroup):
                raise NotImplementedError(f"Not implemented for {fagrp.__class__}.")
```

**Review by inducer — APPROVED** [review]
Thanks! I'll apply the one suggestion.

File: meshmode/mesh/__init__.py (hunk 3 above)
Comment (majosm) [reply meshmode/mesh/__init__.py]: I think this might not work since `InterPartitionAdjacencyGroup` inherits from `FacialAdjacencyGroup`.

**Comment by inducer** [comment]
> majosm disabled auto-merge 5 minutes ago

?

**Comment by majosm** [comment]
> > majosm disabled auto-merge 5 minutes ago
> 
> ?

See comment above.

File: meshmode/mesh/__init__.py (hunk 3 above)
Comment (inducer) [reply meshmode/mesh/__init__.py]: Good catch! 7628622

**Pull request merged**